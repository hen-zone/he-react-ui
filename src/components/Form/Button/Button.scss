@import '../../../styles/base';

.buttonContainer {
  display: inline-flex;
  height: 48px;
  min-width: 180px;

  align-items: center;
  justify-content: center;

  &+.buttonContainer {
    margin-left: 24px;
  }

  @include breakpoint(small) {
    min-width: 0;
    width: 100%;
    &+.buttonContainer {
      margin-left: 0;
      margin-top: 16px;
    }
  }

  &.small {
    height: 30px;
    min-width: 100px;
    @include breakpoint(small) {
      min-width: 0;
    }
  }

  &.large {
    min-width: 280px;
    @include breakpoint(small) {
      min-width: 0;
    }
  }
}

.button {
  display: flex;
  flex-grow: 1;
  height: 100%;
  flex-basis: 48px;
  white-space: nowrap;
  cursor: pointer;
  color: $white;
  padding: 0 36px;
  border: none;
  border-radius: 48px;
  text-align: left;
  position: relative;
  font-size: $input-size;
  font-family: $font-stack;
  outline: none;
  font-weight: $semi-bold;
  text-decoration: none;
  user-select: none;
  align-items: center;
  justify-content: center;
  width: 100%;

  transition-property: flex-grow;
  transition-duration: 0.3s;

  &.submitting {
    flex-grow: 0.01; // working around a chrome bug
    padding: 0;
    pointer-events: none;
  }

  &.done {
    background-color: $green;
    pointer-events: none;
  }

  &:not(.disabled):not(.submitting):not(.done):hover {
    filter: brightness(92%);
  }
  &:not(.disabled):not(.submitting):not(.done):active {
    filter: brightness(84%);
  }
}

.content {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}


.teal {
  background: $teal;
  &:active, &:hover, &:focus {
    background: $teal;
  }
}

.blue {
  background: $blue;
  &:active, &:hover, &:focus {
    background: $blue;
  }
}

.green {
  background: $green;
  &:active, &:hover, &:focus {
    background: $green;
  }
}

.red {
  background: $red;
  &:active, &:hover, &:focus {
    background: $red;
  }
}

.white {
  background: $white;
  color: $teal_dark_20;
  &:active, &:hover, &:focus {
    background: $white;
  }
}

.link {
  background: none;
  border: none;
  padding: 0;
  height: auto;
  &:active, &:hover, &:focus {
    background: none;
  }
}

.link, .keyline:not(:hover):not(:active) {
  border-width: 2px;

  &.teal {
    color: $teal;
    border-color: $teal;
  }

  &.blue {
    color: $blue;
    border-color: $blue;
  }

  &.green {
    color: $green;
    border-color: $green;
  }

  &.red {
    color: $red;
    border-color: $red;
  }

  &.white {
    color: $white;
    border-color: $white;
  }
}

.squared {
  border-radius: 2px;
}

.iconContainer {
  position: absolute;
  height: 24px;
  width: 24px;
  margin-top: -12px;
  top: 50%;

  >* {
    width: 100%;
    height: 100%;
  }
}

.iconLeft {
  @extend .iconContainer;
  left: 12px;
}

.iconRight {
  @extend .iconContainer;
  right: 12px;
}

.iconCenter {
  @extend .iconContainer;
  left: 50%;
  margin-left: -12px;
}

.legacyIcon {
  fill: currentColor;
  height: 24px;
  width: 24px;
  position: absolute;
  right: 12px;
  top: 12px;
}

.keyline {
  &:not(:hover):not(:active) {
    border: solid 1px;
    background: none;
  }
}

.disabled {
  cursor: not-allowed;
  &:not(.link) {
    background: $grey_4 !important;
    &:active, &:hover, &:focus {
      background: $grey_4 !important;
    }
  }
  color: $white !important;
  border: none !important;
}
