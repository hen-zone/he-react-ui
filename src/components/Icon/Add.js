// @flow
import React from 'react';

function Add(props: SVGProps) {
  return (
    <svg {...props} viewBox="0 0 24 24">
      <path d="M10.994 11V6.343a1 1 0 0 1 2 0V11h4.657a1 1 0 0 1 0 2h-4.657v4.657a1 1 0 1 1-2 0V13H6.338a1 1 0 1 1 0-2h4.656z" />
    </svg>
  );
}

Add.defaultProps = {
  height: 32,
  width: 32,
  fill: 'currentColor',
};

export default Add;
