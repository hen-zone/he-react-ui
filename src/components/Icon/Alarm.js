// @flow
import React from 'react';

function Alarm(props: SVGProps) {
  return (
    <svg {...props} viewBox="0 0 24 24">
      <path d="M19.04 8.2a8.005 8.005 0 0 1-1.18 9.246l1.847 1.847a1 1 0 0 1-1.414 1.414l-1.973-1.972a8.01 8.01 0 0 1-8.64 0l-1.973 1.972a1 1 0 1 1-1.414-1.414l1.847-1.847A8.005 8.005 0 0 1 4.958 8.2a3 3 0 1 1 4.573-3.81 8.034 8.034 0 0 1 4.938 0 3 3 0 1 1 4.572 3.81zm-3.366-.944A6.002 6.002 0 0 0 7.29 8.281a6 6 0 1 0 8.383-1.025zM13 11.5h2.5a1 1 0 0 1 0 2H12a1 1 0 0 1-1-1V9a1 1 0 1 1 2 0v2.5zM6.343 6.343c.41-.41.857-.77 1.336-1.077a1 1 0 0 0-1.523 1.27c.061-.065.123-.13.187-.193zm11.313 0c.064.064.126.129.188.194A.999.999 0 0 0 17 5a1 1 0 0 0-.68.266c.48.308.927.668 1.336 1.077z" />
    </svg>
  );
}

Alarm.defaultProps = {
  height: 32,
  width: 32,
  fill: 'currentColor',
};

export default Alarm;
