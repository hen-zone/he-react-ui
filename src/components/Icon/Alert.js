// @flow
import React from 'react';

function Alert(props: SVGProps) {
  return (
    <svg {...props} viewBox="0 0 20 20">
      <path d="M10.882 3.654l6.334 11.875A1 1 0 0 1 16.333 17H3.667a1 1 0 0 1-.883-1.47L9.118 3.653a1 1 0 0 1 1.764 0zM9.25 8v2.5a.75.75 0 0 0 1.5 0V8a.75.75 0 0 0-1.5 0zm.75 7a1 1 0 1 0 0-2 1 1 0 0 0 0 2z" />
    </svg>
  );
}

Alert.defaultProps = {
  height: 32,
  width: 32,
  fill: 'currentColor',
};

export default Alert;
