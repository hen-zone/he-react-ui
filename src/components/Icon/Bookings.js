// @flow
import React from 'react';

function Bookings(props: SVGProps) {
  return (
    <svg {...props} viewBox="0 0 24 24">
      <path d="M.984 7.986A1 1 0 0 0 0 9v14a1 1 0 0 0 1 1h22a1 1 0 0 0 1-1V9a1 1 0 1 0-2 0v13H2V9A1 1 0 0 0 .984 7.986z" />
      <path d="M1 2a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h22a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1zm1 2h20v4H2z" />
      <path d="M11.984-.014A1 1 0 0 0 11 1v4a1 1 0 1 0 2 0V1a1 1 0 0 0-1.016-1.014zM5.984-.014A1 1 0 0 0 5 1v4a1 1 0 1 0 2 0V1A1 1 0 0 0 5.984-.014zM17.984-.014A1 1 0 0 0 17 1v4a1 1 0 1 0 2 0V1a1 1 0 0 0-1.016-1.014zM4 12h4v3H4zM10 12h4v3h-4zM4 17h4v3H4zM10 17h4v3h-4zM16 12h4v3h-4z" />
    </svg>
  );
}

Bookings.defaultProps = {
  height: 32,
  width: 32,
  fill: 'currentColor',
};

export default Bookings;
