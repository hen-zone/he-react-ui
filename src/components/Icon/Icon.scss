@import "../../styles/base";

.icon {
  fill: currentColor;
}

.defaultSize {
  height: 32px;
  width: 32px;
}

.teal {
  border-color: $teal_1;
  color: $teal_1;
}

.blue {
  border-color: $sp_blue;
  color: $sp_blue;
}

.green {
  border-color: $sp_green;
  color: $sp_green;
}

.red {
  border-color: $sp_red;
  color: $sp_red;
}

.white {
  border-color: $white;
  color: $white;
}
