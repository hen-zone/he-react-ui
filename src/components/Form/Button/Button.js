// @flow

/**
 *
 * Button
 *
 */

import classnames from 'classnames';
import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import Icon from '../../Icon';
import styles from './Button.scss';
import ButtonSpinner from './ButtonSpinner';

function InnerButton(props: {
  to: ?string,
  withRouter: boolean,
  submit: boolean,
  children: any,
  className: string,
  onClick: ?Function,
}) {
  const { to, withRouter, submit, children, className, onClick } = props;
  if (!to) {
    return (
      <button
        className={className}
        onClick={onClick}
        type={submit ? 'submit' : 'button'}
      >
        {children}
      </button>
    );
  }

  return withRouter ? (
    <Link to={to} className={className}>
      {children}
    </Link>
  ) : (
    <a href={to} className={className}>
      {children}
    </a>
  );
}

type Props = {
  submit: boolean,
  color?: 'teal' | 'blue' | 'green' | 'red' | 'white',
  link?: boolean,
  small?: boolean,
  large?: boolean,
  onClick: Function,
  keyline?: boolean,
  children?: string | Array<any>,
  disabled?: boolean,
  icon?: string,
  iconLeft?: any,
  iconRight?: any,
  squared?: boolean,
  style?: any,
  submitting?: boolean,
  done?: boolean,
  className?: string,
  to?: string,
  withRouter: boolean,
};

class Button extends React.Component<Props> {
  static defaultProps = {
    submit: false,
    link: false,
    color: 'teal',
    icon: '',
    keyline: false,
    disabled: false,
    squared: false,
    onClick: () => null,
    withRouter: false,
  };

  handleClick = (event: SyntheticEvent<*>) => {
    if (!this.props.disabled) {
      this.props.onClick(event);
    }
  };

  render() {
    const {
      children,
      className,
      color,
      disabled,
      icon,
      iconLeft,
      iconRight,
      keyline,
      link,
      small,
      large,
      squared,
      style,
      submit,
      submitting,
      done,
      to,
      withRouter,
    } = this.props;

    const buttonClasses = classnames(
      styles.button,
      color && !done && styles[color],
      {
        [styles.disabled]: disabled,
        [styles.keyline]: keyline,
        [styles.link]: link,
        [styles.squared]: squared,
        [styles.submitting]: submitting,
        [styles.done]: done,
      },
      className,
    );

    const containerClasses = classnames(styles.buttonContainer, {
      [styles.small]: small,
      [styles.large]: large,
    });

    const submittingIcon = submitting ? (
      <div className={styles.iconCenter}>
        <ButtonSpinner />
      </div>
    ) : null;

    const statusIcon = done ? (
      <div className={styles.iconCenter}>
        <Icon name="Tick" />
      </div>
    ) : (
      submittingIcon
    );

    return (
      <div className={containerClasses} style={style}>
        <InnerButton
          className={buttonClasses}
          submit={submit}
          to={to}
          withRouter={withRouter}
          onClick={this.handleClick}
        >
          {statusIcon || (
            <Fragment>
              {iconLeft && <div className={styles.iconLeft}>{iconLeft}</div>}

              <div className={styles.content}>
                {children}

                {icon && <Icon className={styles.legacyIcon} name={icon} />}
              </div>

              {iconRight && <div className={styles.iconRight}>{iconRight}</div>}
            </Fragment>
          )}
        </InnerButton>
      </div>
    );
  }
}

export default Button;
