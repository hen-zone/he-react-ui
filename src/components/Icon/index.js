// @flow
import classnames from 'classnames';
import React from 'react';
import Add from './Add';
import Alarm from './Alarm';
import Alert from './Alert';
import {
  ArrowLeft,
  ArrowRight,
  ArrowUp,
  ArrowDown,
  CurvedArrow,
} from './Arrows';
import Bookings from './Bookings';
import Calendar from './Calendar';
import CaretRight from './CaretRight';
import {
  CheckBoxChecked,
  CheckBoxUnchecked,
  RadioChecked,
  RadioUnchecked,
} from './CheckBoxes';
import { ChevronDown, ChevronLeft, ChevronRight, ChevronUp } from './Chevrons';
import Clock from './Clock';
import Cross from './Cross';
import Delete from './Delete';
import DropDown from './DropDown';
import Edit from './Edit';
import Ellipsis from './Ellipsis';
import Engage from './Engage';
import HealthEngine from './HealthEngine';
import HealthEngineWithText from './HealthEngineWithText';
import Help from './Help';
import Home from './Home';
import style from './Icon.scss';
import Logout from './Logout';
import Megaphone from './Megaphone';
import Patients from './Patients';
import Search from './Search';
import Settings from './Settings';
import Sync from './Sync';
import Tick from './Tick';
import View from './View';

type Color = 'teal' | 'blue' | 'green' | 'red' | 'white';

function Icon(props: {
  width?: number,
  height?: number,
  color?: Color,
  className?: string,
  name: string,
}) {
  const { width, height, color, className, name, ...restProps } = props;
  const classes = classnames(style.icon, color && style[color], className);
  const childProps = {
    width,
    height,
    className: classes,
    ...restProps,
  };

  switch (name) {
    case 'Add':
      return <Add {...childProps} />;
    case 'Alarm':
      return <Alarm {...childProps} />;
    case 'Alert':
      return <Alert {...childProps} />;
    case 'ArrowLeft':
      return <ArrowLeft {...childProps} />;
    case 'ArrowRight':
      return <ArrowRight {...childProps} />;
    case 'ArrowUp':
      return <ArrowUp {...childProps} />;
    case 'ArrowDown':
      return <ArrowDown {...childProps} />;
    case 'Bookings':
      return <Bookings {...childProps} />;
    case 'Calendar':
      return <Calendar {...childProps} />;
    case 'CaretRight':
      return <CaretRight {...childProps} />;
    case 'CheckBoxChecked':
      return <CheckBoxChecked {...childProps} />;
    case 'CheckBoxUnchecked':
      return <CheckBoxUnchecked {...childProps} />;
    case 'ChevronLeft':
      return <ChevronLeft {...childProps} />;
    case 'ChevronRight':
      return <ChevronRight {...childProps} />;
    case 'ChevronUp':
      return <ChevronUp {...childProps} />;
    case 'ChevronDown':
      return <ChevronDown {...childProps} />;
    case 'Clock':
      return <Clock />;
    case 'CurvedArrow':
      return <CurvedArrow />;
    case 'Cross':
      return <Cross {...childProps} />;
    case 'Delete':
      return <Delete {...childProps} />;
    case 'DropDown':
      return <DropDown {...childProps} />;
    case 'Edit':
      return <Edit {...childProps} />;
    case 'Engage':
      return <Engage {...childProps} />;
    case 'HealthEngine':
      return <HealthEngine {...restProps} />;
    case 'HealthEngineWithText':
      return <HealthEngineWithText />;
    case 'Help':
      return <Help {...childProps} />;
    case 'Home':
      return <Home {...childProps} />;
    case 'Megaphone':
      return <Megaphone {...childProps} />;
    case 'Patients':
      return <Patients {...childProps} />;
    case 'Ellipsis':
      return <Ellipsis {...childProps} />;
    case 'Logout':
      return <Logout {...childProps} />;
    case 'RadioChecked':
      return <RadioChecked {...childProps} />;
    case 'RadioUnchecked':
      return <RadioUnchecked {...childProps} />;
    case 'Search':
      return <Search {...childProps} />;
    case 'Settings':
      return <Settings {...childProps} />;
    case 'Sync':
      return <Sync {...childProps} />;
    case 'Tick':
      return <Tick {...childProps} />;
    case 'View':
      return <View {...childProps} />;
    default:
      return <Cross {...childProps} />;
  }
}

Icon.defaultProps = {
  name: 'Cross',
};

export default Icon;
