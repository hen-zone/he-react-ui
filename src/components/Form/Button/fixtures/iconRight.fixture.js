import React from 'react'
import Button from '../'
import Icon from '../../../Icon'

export default {
  component: Button,
  props: {
    iconRight: <Icon name="ArrowRight" />
  },
  children: 'Button',
  simulateSubmission: true
}
