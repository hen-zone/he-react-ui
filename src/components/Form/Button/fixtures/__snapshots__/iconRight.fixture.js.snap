// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anonymous 1`] = `
<Button
  color="teal"
  disabled={false}
  done={false}
  icon=""
  iconRight={
    <Icon
      name="ArrowRight"
    />
  }
  keyline={false}
  link={false}
  newWindow={false}
  onClick={[Function]}
  squared={false}
  submit={false}
  submitting={false}
  withRouter={false}
>
  <div
    className="buttonContainer"
  >
    <InnerButton
      className="button teal"
      newWindow={false}
      onClick={[Function]}
      submit={false}
      withRouter={false}
    >
      <button
        className="button teal"
        onClick={[Function]}
        type="button"
      >
        <div
          className="content"
        >
          Button
        </div>
        <div
          className="iconRight"
        >
          <Icon
            name="ArrowRight"
          >
            <ArrowRight
              className="icon"
            >
              <svg
                className="icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.586 13H3a1 1 0 1 1 0-2h15.586l-4.293-4.293a1 1 0 1 1 1.414-1.414l6 6a1 1 0 0 1 0 1.414l-6 6a1 1 0 0 1-1.414-1.414L18.586 13z"
                />
              </svg>
            </ArrowRight>
          </Icon>
        </div>
      </button>
    </InnerButton>
  </div>
</Button>
`;
